components/mpas-ocean/src/shared/mpas_ocn_submesoscale_eddies.F

```diff
@@ -0,0 +1,182 @@
+  !>  min(dcEdge,dsMax), H is the mixed layer depth, L_f is the subgrid
+  !>  frontal width that is scaled to the coarse grid, f is the 
+  !>  coriolis parameter, and tau is a time scale specified to prevent 
+  !>  the singularity near the equator.  the horizontal buoyancy gradient
+  !>  is integrated over the mixed layer depth and mu(z) is a non-dimensional
+  !>  shape function that is zero below the mixed layer depth
+  !>
+  !***********************************************************************
+
+  subroutine ocn_submesoscale_compute_velocity()
+
+     !-----------------------------------------------------------------
+     !
+     ! local variables
+     !
+     !-----------------------------------------------------------------
+
+     integer :: k, nEdges, nCells, iCell, iEdge, cell1, cell2, indMLDedge
+
+     real(kind=RKIND),dimension(:),allocatable :: &
+        streamFunction, &
+        zEdge
+
+     real(kind=RKIND) :: mu, zMLD, bvfML, hML, gradBuoyML, ds, Lf, bvfAv, &
+        hAv, bldEdge, ustarEdge
+
+     nEdges = nEdgesHalo(2)
+
+     allocate(streamFunction(nVertLevels+1))
+     allocate(zEdge(nVertLevels+1))
+
+     !$omp parallel
+     !$omp do schedule(runtime) &
+     !$omp private(cell1,cell2,bvfML,zEdge,streamFunction,gradBuoyML,hML,bvfAv, &
+     !$omp         zMLD,Lf,ds,mu,k)
+     do iEdge=1,nEdges
+        cell1 = cellsOnEdge(1,iEdge)
+        cell2 = cellsOnEdge(2,iEdge)
+
+        zEdge(:) = 0.0_RKIND
+        streamFunction(:) = 0.0_RKIND
+        bvfML = 0.0_RKIND
+        gradBuoyML = 0.0_RKIND
+        bvfAv = sqrt(0.5_RKIND*(max(BruntVaisalaFreqTop(minLevelCell(cell1)+1,cell1),1.0E-20_RKIND) + &
+                       max(BruntVaisalaFreqTop(minLevelCell(cell2)+1,cell2),1.0E-20_RKIND)))
+        hML = 0.5_RKIND*layerThickEdgeMean(minLevelEdgeBot(iEdge),iEdge)
+        bvfML = bvfML + hML*bvfAv
+        if (minLevelEdgeTop(iEdge) .ge. 1) then
+           gradBuoyML = hML*gradBuoyEddy(minLevelEdgeTop(iEdge)+1,iEdge)
+           bvfML = hML*bvfAv
+        else
+           cycle
+        end if
+
+        indMLDedge = min(indMLD(cell1),indMLD(cell2))
+        do k = minLevelEdgeBot(iEdge)+2,indMLDedge
+           hAv = 0.5_RKIND*(layerThickEdgeMean(k,iEdge) + layerThickEdgeMean(k-1,iEdge))
+           bvfML = bvfML + hAv*bvfAv
+           hML = hML + hAv
+           bvfAv = sqrt(0.5_RKIND*(max(BruntVaisalaFreqTop(k,cell1),1.0E-20_RKIND) + &
+                              max(BruntVaisalaFreqTop(k,cell2),1.0E-20_RKIND)))
+           gradBuoyML = gradBuoyML + hAv*gradBuoyEddy(k,iEdge)
+        end do
+        bvfML = bvfML / (1.0E-20_RKIND + hML)
+        gradBuoyML = gradBuoyML / (1.0E-20_RKIND + hML)
+
+        !compute depths and shape function
+
+        do k = minLevelEdgeTop(iEdge)+1,maxLevelEdgeTop(iEdge)+1
+           zEdge(k) = zedge(k-1) - layerThickEdgeMean(k-1,iEdge)
+         end do
+
+        zMLD = 0.5_RKIND*(dThreshMLD(cell1)+dThreshMLD(cell2))
```

Comment:
Can someone clarifies what happens near topography/land? 
My understanding is that near topo, we will get dThreshMLD(iCell)=abs(max(mldTemp,depth(k))) so the zMLD value will be 0.5*(dThreshMLD(validcell)+ depth of other cell), which makes sense to me. 
What happens at land boundaries? is dThreshMLD zero in the land cell, or is the zMLD not computed on ocean-land edges? 
(maybe that's what nEdgesHalo(2) does but I can't remember off the top of my head)

Reply:
if you look up at L109 you'll see that when you are by topography (minLevelEdgeTop = 0) it doesn't do the calculation.  So this line is never encountered at the coastline

Reply:
Ok, thanks for clarifying!